src/Algebra/Properties/Semiring/Mult.agda

```diff
@@ -60,3 +55,32 @@ open import Algebra.Properties.Monoid.Mult +-monoid public
   x * y + (n × x) * y   ≈⟨ +-congˡ (×-assoc-* n _ _) ⟩
   x * y + n × (x * y)   ≡⟨⟩
   suc n × (x * y)       ∎
+
+-- (_× x) is homomorphic with respect to _ℕ.*_/_*_ for idempotent x.
+
+idem-×-homo-* : ∀ m n {x} → (_*_ IdempotentOn x) → (m × x) * (n × x) ≈ (m ℕ.* n) × x
+idem-×-homo-* m n {x} idem = begin
+  (m × x) * (n × x)   ≈⟨ ×-assoc-* m x (n × x) ⟩
+  m × (x * (n × x))   ≈⟨ ×-congʳ m (×-comm-* n x x) ⟩
+  m × (n × (x * x))   ≈⟨ ×-assocˡ _ m n ⟩
+  (m ℕ.* n) × (x * x) ≈⟨ ×-congʳ (m ℕ.* n) idem ⟩
+  (m ℕ.* n) × x       ∎
+
+-- (_× 1#) is homomorphic with respect to _ℕ.*_/_*_.
+
+×1-homo-* : ∀ m n → (m ℕ.* n) × 1# ≈ (m × 1#) * (n × 1#)
+×1-homo-* m n = sym (idem-×-homo-* m n (*-identityʳ 1#))
+
+------------------------------------------------------------------------
+-- DEPRECATED NAMES
+------------------------------------------------------------------------
+-- Please use the new names as continuing support for the old names is
+-- not guaranteed.
+
+-- Version 2.1
+
+1×-identityʳ = ×-homo-1
+{-# WARNING_ON_USAGE 1×-identityʳ
+"Warning: 1×-identityʳ was deprecated in v2.1.
+Please use Algebra.Properties.Monoid.Mult.×-homo-1 instead. "
```

Comment:
Given that `×-homo-1` is already publicly exported by this module, I don't think the name should be qualified in the warning.